src/chunk.c

```diff
@@ -900,7 +900,9 @@ chunk_set_replica_identity(const Chunk *chunk)
 	}
 
 	ts_catalog_database_info_become_owner(ts_catalog_database_info_get(), &sec_ctx);
+	EventTriggerAlterTableStart((Node *) &cmd);
 	AlterTableInternal(chunk->table_id, list_make1(&cmd), false);
+	EventTriggerAlterTableEnd();
 	ts_catalog_restore_user(&sec_ctx);
 	table_close(ht_rel, NoLock);
```

Comment:
This can be called in a nested fashion since the start and end commands remember the previous value of `currentCommand`. You might want to add a comment explaining why the start and end are necessary though, for your future self.

Reply:
Actually we have a function `ts_alter_table_with_event_trigger` to wrap the event trigger context and execute the `AlterTableInternal` that have the enough comments about why it is necessary.